src/irr/video/COpenGLSpecializedShader.cpp

```diff
@@ -204,14 +204,18 @@ void COpenGLSpecializedShader::buildUniformsList()
             for (size_t i = 0ull; i < top.members.count; ++i) {
                 SMember m = top.members.array[i];
                 m.name = top.name + "." + m.name;
+                if (m.count > 1u)
+                    m.name += "[0]";
                 q.push(m);
             }
             continue;
         }
         using gl = COpenGLExtensionHandler;
         const GLint location = gl::extGlGetUniformLocation(m_GLname, top.name.c_str());
+        assert(location != -1);
         m_uniformsList.emplace_back(top, location);
     }
+    std::sort(m_uniformsList.begin(), m_uniformsList.end(), [](const SUniform& a, const SUniform& b) { return a.location < b.location; });
```

Comment:
why is this happenning every time we want to set push constants?

Reply:
it's not, it's in `buildUniformList()`

Reply:
ah ok github messed up